### PR TITLE
fix: fix verifiable credential payload

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
## Description
this PR is to fix the vc utils to output the same structure as veramo sdk

e.g output from veramo sdk

```json
 {
   "issuer":{
      "id":"did:pkh:eip155:1:0xcAB5742Cf80C6C4EeCa9CBA3F509Ba6875D05671"
   },
   "@context":[
      "https://www.w3.org/2018/credentials/v1",
      "https://veramo.io/contexts/profile/v1"
   ],
   "type":[
      "VerifiableCredential",
      "TrustCredential"
   ],
   "issuanceDate":"2024-01-27T06:44:31.083Z",
   "credentialSubject":{
      "id":"did:pkh:eip155:1:0xcAB5742Cf80C6C4EeCa9CBA3F509Ba6875D05671",
      "name ":"alice"
   },
   "proof":{
      "verificationMethod":"did:pkh:eip155:1:0xcAB5742Cf80C6C4EeCa9CBA3F509Ba6875D05671#blockchainAccountId",
      "created":"2024-01-27T06:44:31.083Z",
      "proofPurpose":"assertionMethod",
      "type":"EthereumEip712Signature2021",
      "proofValue":"0x7787e7323c7168818ec25dc3ea2e438bd39ba18505e6864f4737cb207f93bafc4e29f616b60de0d8e1c0066cbd0dbb54e5360c9a360ff7c9fed900cea1dd31f01c",
      "eip712":{
         "domain":{
            "chainId":1,
            "name":"VerifiableCredential",
            "version":"1"
         },
         "types":{
            "EIP712Domain":[
               {
                  "name":"name",
                  "type":"string"
               },
               {
                  "name":"version",
                  "type":"string"
               },
               {
                  "name":"chainId",
                  "type":"uint256"
               }
            ],
            "CredentialSubject":[
               {
                  "name":"id",
                  "type":"string"
               },
               {
                  "name":"name ",
                  "type":"string"
               }
            ],
            "Issuer":[
               {
                  "name":"id",
                  "type":"string"
               }
            ],
            "Proof":[
               {
                  "name":"created",
                  "type":"string"
               },
               {
                  "name":"proofPurpose",
                  "type":"string"
               },
               {
                  "name":"type",
                  "type":"string"
               },
               {
                  "name":"verificationMethod",
                  "type":"string"
               }
            ],
            "VerifiableCredential":[
               {
                  "name":"@context",
                  "type":"string[]"
               },
               {
                  "name":"credentialSubject",
                  "type":"CredentialSubject"
               },
               {
                  "name":"issuanceDate",
                  "type":"string"
               },
               {
                  "name":"issuer",
                  "type":"Issuer"
               },
               {
                  "name":"proof",
                  "type":"Proof"
               },
               {
                  "name":"type",
                  "type":"string[]"
               }
            ]
         },
         "primaryType":"VerifiableCredential"
      }
   }
}
```

e.g output from the utils
```json
{
    "@context": [
        "https://www.w3.org/2018/credentials/v2"
    ],
    "type": [
        "VerifiableCredential",
        "TrustCredential"
    ],
    "issuanceDate": "2024-01-29T07:34:02.428Z",
    "issuer": "did:pkh:eip155:59144:0x3892967AA898d7EeBf1B08d3E1F31B2F4C84317A",
    "credentialSubject": {
        "id": "did:pkh:eip155:59144:0x6B24aE0ABbeb67058D07b891aF415f288eA57Cc7",
        "trustworthiness": [
            {
                "scope": "Software development",
                "level": 0,
                "reason": []
            },
            {
                "scope": "Software security",
                "level": 1,
                "reason": []
            }
        ]
    },
    "proof": {
        "type": "EthereumEip712Signature2021",
        "proofValue": "0xe4634796f29d983bbc88e74dc8d5e5fd49d12046276da459a830f75386c7b0a35d212abe639d89e8f48ca77228c307e637fd7b47703b4fd7068358545734b9dc1b",
        "proofPurpose": "assertionMethod",
        "verificationMethod": "did:pkh:eip155:59144:0x3892967AA898d7EeBf1B08d3E1F31B2F4C84317A#blockchainAccountId",
        "created": "2024-01-29T07:34:05.640Z",
        "eip712": {
            "domain": {
                "name": "VerifiableCredential",
                "version": "1",
                "chainId": 59144
            },
            "types": {
                "EIP712Domain": [
                    {
                        "name": "name",
                        "type": "string"
                    },
                    {
                        "name": "version",
                        "type": "string"
                    },
                    {
                        "name": "chainId",
                        "type": "uint256"
                    }
                ],
                "VerifiableCredential": [
                    {
                        "name": "@context",
                        "type": "string[]"
                    },
                    {
                        "name": "type",
                        "type": "string[]"
                    },
                    {
                        "name": "issuer",
                        "type": "string"
                    },
                    {
                        "name": "credentialSubject",
                        "type": "CredentialSubject"
                    },
                    {
                        "name": "issuanceDate",
                        "type": "string"
                    },
                    {
                        "name": "proof",
                        "type": "Proof"
                    }
                ],
                "Trustworthiness": [
                    {
                        "name": "scope",
                        "type": "string"
                    },
                    {
                        "name": "level",
                        "type": "int8"
                    },
                    {
                        "name": "reason",
                        "type": "string[]"
                    }
                ],
                "CredentialSubject": [
                    {
                        "name": "id",
                        "type": "string"
                    },
                    {
                        "name": "trustworthiness",
                        "type": "Trustworthiness[]"
                    }
                ],
                "Proof": [
                    {
                        "name": "created",
                        "type": "string"
                    },
                    {
                        "name": "proofPurpose",
                        "type": "string"
                    },
                    {
                        "name": "type",
                        "type": "string"
                    },
                    {
                        "name": "verificationMethod",
                        "type": "string"
                    }
                ]
            },
            "primaryType": "VerifiableCredential"
        }
    }
}
```
### Related ticket

Fixes #

### Type of change

- [ ] Chore
- [X] Bug fix
- [ ] New feature
- [ ] Documentation update

### CODE REVIEW

**Review by ritave — APPROVED** [review]

File: src/utils/verifiable-credential/baseVC.ts [hunk 1]
```diff
@@ -49,44 +97,46 @@ export abstract class BaseVerifiableCredential {
...
+      type: 'Proof',
+    });
+
     return {
       type: 'EthereumEip712Signature2021',
       proofValue: signature,
       proofPurpose: 'assertionMethod',
+      verificationMethod: `${
+        (signArg.message as Assertion).issuer
+      }#blockchainAccountId`,
```
Comment (dawnseeker8) [inline src/utils/verifiable-credential/baseVC.ts:129]: Do you mean that `verificationMethod` here is issuer blockchainAccountId, which will be hash of DID of issuer?

**Review by dawnseeker8 — APPROVED** [review]
LGTM,

File: src/utils/verifiable-credential/baseVC.ts (hunk 1 above)
Comment (stanleyyconsensys) [reply src/utils/verifiable-credential/baseVC.ts:129]: actually, not sure yet, still waiting K3L to reply us where it is correct or not ,as they are the one who validate the message

File: src/utils/verifiable-credential/types.ts
Comment (dawnseeker8) [inline src/utils/verifiable-credential/types.ts:1]: regarding @dayksx requirement to change `StatusCredential` to `ReviewCredential`, please change the `TrustCredentialType` 
```
export enum TrustCredentialType {
 ...
  ReviewCredential = 'ReviewCredential',
}
```

File: src/utils/verifiable-credential/snapVC.test.ts [hunk 2]
```diff
@@ -1,78 +1,83 @@
...
     it('builds a valid verifiable credential', () => {
       const vc = buildSnapVerifiableCredential();
       expect(
         vc.buildEndosedPayload(VALID_ACCOUNT_1, SNAP_SHASUM_1, ['reason']),
       ).toStrictEqual({
-        domain,
+        domain: typedDataDomain,
         message: {
           '@context': ['https://www.w3.org/2018/credentials/v2'],
           type: ['VerifiableCredential', 'StatusCredential'],
```
Comment (dawnseeker8) [inline src/utils/verifiable-credential/snapVC.test.ts:78]: Need to change `StatusCredential` to `ReviewCredential`

File: src/utils/verifiable-credential/snapVC.test.ts [hunk 3]
```diff
@@ -94,11 +99,12 @@ describe('SnapVerifiableCredential', () => {
       expect(
         vc.buildDisputedPayload(VALID_ACCOUNT_1, SNAP_SHASUM_1, ['reason']),
       ).toStrictEqual({
-        domain,
+        domain: typedDataDomain,
         message: {
           '@context': ['https://www.w3.org/2018/credentials/v2'],
           type: ['VerifiableCredential', 'StatusCredential'],
```
Comment (dawnseeker8) [inline src/utils/verifiable-credential/snapVC.test.ts:105]: Need to change `StatusCredential` to `ReviewCredential`
```suggestion
          type: ['VerifiableCredential', 'ReviewCredential'],
```

File: src/utils/verifiable-credential/snapVC.test.ts (hunk 2 above)
Comment (dawnseeker8) [inline src/utils/verifiable-credential/snapVC.test.ts:78]: Need to change `StatusCredential` to `ReviewCredential`
```suggestion
          type: ['VerifiableCredential', 'ReviewCredential'],
```

**Review by alainncls — APPROVED** [review]

**Pull request merged**